src/main/java/bio/terra/service/filedata/FileService.java

```diff
@@ -177,6 +177,49 @@ public String ingestBulkFileArray(
         .submit();
   }
 
+  public List<FileModel> listDatasetFiles(String datasetId, int offset, int limit) {
+    Dataset dataset = datasetService.retrieve(UUID.fromString(datasetId));
+    CloudPlatformWrapper cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    if (cloudPlatformWrapper.isGcp()) {
+      try {
+        return fileDao.batchRetrieveFiles(dataset, dataset, offset, limit);
+      } catch (InterruptedException ex) {
+        throw new FileSystemExecutionException(ex);
+      }
+    } else {
+      String collectionId = DATASET.toTableName(dataset.getId());
+      AzureStorageAuthInfo storageAuthInfo = resourceService.getDatasetStorageAuthInfo(dataset);
+      return tableDao.batchRetrieveFiles(
+          collectionId, storageAuthInfo, datasetId, storageAuthInfo, offset, limit);
+    }
+  }
+
+  public List<FileModel> listSnapshotFiles(String snapshotId, Integer offset, Integer limit) {
```

Comment:
I think these should also use unboxed types
```suggestion
  public List<FileModel> listSnapshotFiles(String snapshotId, int offset, int limit) {
```